src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt

```diff
@@ -256,57 +252,35 @@ class ManagedIndexCoordinator(
-        val indicesToRemoveManagedIndexMetaDataFrom = mutableListOf<Index>()
-        event.state().metadata().indices().forEach {
-            val previousIndexMetaData = event.previousState().metadata().index(it.value.index)
-            val policyID = it.value.getPolicyID()
-            val request: DocWriteRequest<*>? = when {
-                it.value.shouldCreateManagedIndexConfig(previousIndexMetaData) && policyID != null -> {
-                    indicesWithPolicyID.add(it.value.index.name)
-                    hasCreateRequests = true
-                    managedIndexConfigIndexRequest(it.value.index.name, it.value.indexUUID, policyID, jobInterval)
-                }
-                it.value.shouldDeleteManagedIndexConfig(previousIndexMetaData) ->
-                    deleteManagedIndexRequest(it.value.indexUUID)
-                else -> null
-            }
-
-            if (request != null) updateManagedIndicesRequests.add(request)
-
-            if (it.value.shouldDeleteManagedIndexMetaData()) indicesToRemoveManagedIndexMetaDataFrom.add(it.value.index)
+        // deal with indices delete event
+        var removeManagedIndexReq = emptyList<DocWriteRequest<*>>()
+        var indicesToClean = emptyList<Index>()
+        if (event.indicesDeleted().isNotEmpty()) {
+            val deletedIndices = event.indicesDeleted().map { Index(it.name, it.uuid) }
+            val managedState = getManagedIndex(deletedIndices.map { it.uuid })
+            indicesToClean = deletedIndices.filter { it.uuid in managedState.keys }
+            removeManagedIndexReq = indicesToClean.map { deleteManagedIndexRequest(it.uuid) }
         }
 
         // Check if newly created indices matching any ISM templates
         var updateMatchingIndexReq = emptyList<DocWriteRequest<*>>()
         // filter out indices with policyID, they will be picked up in previous block
-        val indicesCreated = event.indicesCreated().filter { it !in indicesWithPolicyID }
-        if (indicesCreated.isNotEmpty()) // only check template match if there are new created indices
-            updateMatchingIndexReq = getMatchingIndicesUpdateReq(event.state(), indicesCreated)
-        if (updateMatchingIndexReq.isNotEmpty()) hasCreateRequests = true
+        if (event.indicesCreated().isNotEmpty()) // only check template match if there are new created indices
+            updateMatchingIndexReq = getMatchingIndicesUpdateReq(event.state(), event.indicesCreated())
 
-        updateManagedIndices(updateManagedIndicesRequests + updateMatchingIndexReq + indicesDeletedRequests, hasCreateRequests)
-        clearManagedIndexMetaData(indicesToRemoveManagedIndexMetaDataFrom)
+        updateManagedIndices(updateMatchingIndexReq + removeManagedIndexReq, updateMatchingIndexReq.isNotEmpty())
+        // TODO currently metadata is auto cleaned after indices deleted
+        //  uncomment this after move metadata to config index
+        // clearManagedIndexMetaData(indicesToClean)
```

Comment:
Confused on this, is this just code from future PR? i.e. won't work now but will once the other PR is added?

Reply:
I notice this when developing. Since this sweepClusterChangedEvent deal with indicesDeleted, currently we still save metadata with index, so index delete means metadata delete also. But when we move metadata to config index, we will need this. So I just leave here as a reminder.